src/Compilers/CSharp/Portable/Lowering/ClosureConversion/ClosureConversion.cs

```diff
@@ -1597,17 +1598,23 @@ private BoundNode RewriteLambdaConversion(BoundLambda node)
                         // different from the local variable `type`, which has the node's type substituted for the current container.
                         var cacheVariableType = containerAsFrame.TypeMap.SubstituteType(node.Type).Type;
 
-                        var cacheVariableName = GeneratedNames.MakeLambdaCacheFieldName(
-                            // If we are generating the field into a display class created exclusively for the lambda the lambdaOrdinal itself is unique already, 
-                            // no need to include the top-level method ordinal in the field name.
-                            (closureKind == ClosureKind.General) ? -1 : topLevelMethodId.Ordinal,
-                            topLevelMethodId.Generation,
-                            lambdaId.Ordinal,
-                            lambdaId.Generation);
-
-                        var cacheField = new SynthesizedLambdaCacheFieldSymbol(translatedLambdaContainer, cacheVariableType, cacheVariableName, _topLevelMethod, isReadOnly: false, isStatic: closureKind == ClosureKind.Singleton);
-                        CompilationState.ModuleBuilderOpt.AddSynthesizedDefinition(translatedLambdaContainer, cacheField);
-                        cache = F.Field(receiver, cacheField.AsMember(constructedFrame)); //NOTE: the field was added to the unconstructed frame type.
+                        // We cannot reference a type argument from the parent method outside the method
+                        // so if the lambda has such an argument we cannot cache it in a field
+                        if (!HasTypeArgumentsFromReferencedMethod(cacheVariableType))
```

Comment:
>HasTypeArgumentsFromReferencedMethod [](start = 29, length = 36)

It feels like the helper you are looking for is ```public static bool ContainsTypeParameter(this TypeSymbol type, MethodSymbol parameterContainer)``` from TypeSymbolExtensions.

Reply:
Wow, awesome! That's the method I was looking for. Thanks for the hint.